test-series/4/tools/sim_prep.py

```diff
@@ -0,0 +1,513 @@
+import copy
+import re
+import numpy as np
+from scipy import interpolate
+
+from pyuvdata import UVData
+from pyuvdata.utils import polstr2num
+from hera_cal.io import to_HERAData
+from hera_cal.utils import lst_rephase
+from hera_sim import noise, Simulator
+if hera_sim.version.startswith('0'):
+    from hera_sim.rfi import _listify
+else:
+    from hera_sim.utils import _listify
+
+
+def add_noise(sim, Trx=100, seed=1010):
+    """
+    Add thermal noise to a UVData object, based on its auto-correlations.
+
+    Parameters
+    ----------
+    sim : :class:`pyuvdata.UVData` or :class:`hera_sim.Simulator` instance
+        The simulation data to add noise to
+    Trx : float, optional
+        The receiver temperature, in Kelvin.
+    seed : int, optional
+        The random seed.
+    """
+    if not isinstance(sim, Simulator):
+        sim = Simulator(data=sim)
+
+    lsts = np.unique(sim.data.lst_array)
+    freqs = np.unique(sim.data.freq_array)
+
+    # Find out which antenna has the autocorrelation data, since we'll
+    # add noise before inflating and adding other systematics.
+    antpair = list(
```

Comment:
Don't need the `list` here.

Reply:
Are you sure? I've had issues in the past with trying to access an individual element from a generator object, so I erred on the side of caution and made sure that it's a list rather than a generator.

Reply:
Ah yeah, didn't see the access of the first element there. Really feels like there should be a better way to do this...

Reply:
It would be nice if there was a simpler way to do it, but this is the best approach I can think of that will work with no prior knowledge of antenna numbers or which antennas actually have data associated with them.

Reply:
Looks like the "correct" answer here is to do

```
next(ants for ants in sim.data.get_antpairs() if ants[0] == ants[1])
```

Reply:
Ah, clever!